staging/src/k8s.io/apiserver/pkg/server/resourceconfig/helpers.go

```diff
@@ -141,22 +167,43 @@ func MergeAPIResourceConfigs(
 		if err != nil {
 			return nil, err
 		}
-		if enabled {
+
+		switch len(tokens) {
+		case 2:
+			versionPreferences = append(versionPreferences, versionEnablementPreference{
+				key:          key,
+				enabled:      enabled,
+				groupVersion: groupVersion,
+			})
+		case 3:
+			resourcePreferences = append(resourcePreferences, resourceEnablementPreference{
+				key:                  key,
+				enabled:              enabled,
+				groupVersionResource: groupVersion.WithResource(tokens[2]),
+			})
+		}
+	}
+
+	// apply version preferences first, so that we can remove the hardcoded resource preferences that are being overridden
+	for _, versionPreference := range versionPreferences {
+		// if a user has expressed a preference about a version, that preference takes priority over the hardcoded resources
+		resourceConfig.RemoveMatchingResourcePreferences(resourceMatcherForVersion(versionPreference.groupVersion))
+		if versionPreference.enabled {
 			// enable the groupVersion for "group/version=true" and "group/version/resource=true"
-			resourceConfig.EnableVersions(groupVersion)
-		} else if len(tokens) == 2 {
+			resourceConfig.EnableVersions(versionPreference.groupVersion)
+
+		} else {
 			// disable the groupVersion only for "group/version=false", not "group/version/resource=false"
```

Comment:
the `not "group/version/resource=false"` portion of this comment no longer applies here